<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="UTF-8">
    <title>Biocompass by NP-Omix</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" type="text/css" href="stylesheets/normalize.css" media="screen">
    <link href='https://fonts.googleapis.com/css?family=Open+Sans:400,700' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
  </head>
  <body>
    <section class="page-header">
      <h1 class="project-name">Biocompass</h1>
      <h2 class="project-tagline">A genome-guided secondary metabolite discovery</h2>
      <a href="https://github.com/NP-Omix/BioCompass" class="btn">View on GitHub</a>
      <a href="https://github.com/NP-Omix/BioCompass/zipball/master" class="btn">Download .zip</a>
      <a href="https://github.com/NP-Omix/BioCompass/tarball/master" class="btn">Download .tar.gz</a>
    </section>

    <section class="main-content">
      <h3>
<a id="welcome-to-biocompass-website" class="anchor" href="#welcome-to-biocompass-website" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Welcome to BioCompass website.</h3>

<p>One of the problems of using network approaches at the field of natural products concerns the concept of networking itself. For accurate dereplication of families (both molecular families, like used in GnPS, and gene cluster families), it’s required to define a threshold for which once trespassed, two gene clusters are not part of the same family anymore. Analogously to species definition via 16S rRNA gene, this threshold is empirical and can be imprecise in some cases (reference here). Hence, BioCompass envision to implement a cutoff calibration feature to minimize this issue. This new feature consists on the user evaluating the network diagram for both gene homology (scored via multigene blast) and domain homology (score via Jaccard index, another feature to be implemented soon), visually deciding which cutoff would better represent those scores for his query. The user will count with an internal standard to aid its decision making process.
</p>

<h3>
<a id="authors-and-contributors" class="anchor" href="#authors-and-contributors" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Authors and Contributors</h3>

<p>You can mention <a href="https://github.com/tiagolbiotech" class="user-mention">@tiagolbiotech</a> or <a href="https://github.com/castelao" class="user-mention">@castelao</a> to generate a link to our profiles.</p>

<h3>
<a id="documentation" class="anchor" href="#documentation" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Documentation</h3>

<p>Check our <a href="http://BioCompass.readthedocs.io/en/latest/">documentation</a> or our manuscript Leão et. al. 2016 (submitted).</p>

      <footer class="site-footer">
        <span class="site-footer-owner"><a href="https://github.com/NP-Omix/BioCompass">Biocompass</a> is maintained by <a href="https://github.com/NP-Omix">NP-Omix</a>.</span>
      </footer>

    </section>

  
  </body>
</html>
